winit-wayland/src/window/mod.rs

```diff
@@ -103,17 +111,91 @@ impl Window {
             WindowDecorations::RequestClient
         };
 
-        let window =
-            state.xdg_shell.create_window(surface.clone(), default_decorations, &queue_handle);
-
-        let mut window_state = WindowState::new(
-            event_loop_window_target.handle.clone(),
-            &event_loop_window_target.queue_handle,
-            &state,
-            size,
-            window.clone(),
-            attributes.preferred_theme,
-        );
+        let (window, mut window_state) = if wl_attrs
+            .as_ref()
+            .is_some_and(|wl_attrs| wl_attrs.is_layer_shell)
+            || attributes.window_level != WindowLevel::Normal
+        {
+            let Some(layer_shell) = state.layer_shell.as_ref() else {
+                return Err(RequestError::NotSupported(NotSupportedError::new(
+                    "No LayerShell exists in this environment",
+                )));
+            };
+            let wl_attrs = wl_attrs.unwrap();
+            let output = wl_attrs.output.and_then(|id| {
+                monitors
+                    .lock()
+                    .unwrap()
+                    .iter()
+                    .find(|m| m.native_id() == id)
+                    .map(|m| m.proxy.clone())
+            });
+            let layer_surface = layer_shell.create_layer_surface(
+                &queue_handle,
+                surface.clone(),
+                wl_attrs.layer.unwrap_or_else(|| layer_from_window_level(attributes.window_level)),
+                Some(attributes.title.clone()),
```

Comment:
The namespace of layer shell is more close to the app id than to the title. E.g. on tiling window managers like Hyprland it is used for matching effects to the surface so it should be more machine readable. IMO it should be the separate attribute not tied to title

Reply:
@Exidex added, let me know what do you think